### PR TITLE
Update to .net 6

### PR DESCRIPTION
**Pull request by vabka**
Use file-scoped namespace everywhere in solution
Add .net5.0 and .net6.0 as targetframeworks
Set langversion to 10

Use some new C# 8 and 9 features (like using-var)

GrokItem could be a record

### CODE REVIEW

**Comment by Marusyk** [comment]
Hi @vabka , thank you for submitting this PR. Will review and update.

File: src/Grok.Net/Grok.Net.csproj [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
-    <Authors>RMarusyk</Authors>
-    <Description>Cross platform .NET grok implementation</Description>
-    <License>https://github.com/Marusyk/grok.net/blob/main/LICENSE</License>
-    <PackageLicenseExpression>MIT</PackageLicenseExpression>
-    <PackageIconUrl>https://raw.githubusercontent.com/Marusyk/grok.net/main/Grok.png</PackageIconUrl>
-    <PackageProjectUrl>https://github.com/Marusyk/grok.net</PackageProjectUrl>
-    <RepositoryUrl>https://github.com/Marusyk/grok.net</RepositoryUrl>
-    <PackageTags>grok;grok-parser;grok-patterns;grokking;c-sharp-library</PackageTags>
-  </PropertyGroup>
+    <PropertyGroup>
```
Comment (Marusyk) [inline src/Grok.Net/Grok.Net.csproj:3]: Please don't add more spaces. XML files should use indent size equals to 2

File: src/Grok.Net/Grok.cs [hunk 2]
```diff
@@ -6,207 +6,205 @@
...
-        private static readonly Regex _grokWithoutName = new Regex("%{(\\w+)}", RegexOptions.Compiled);
+        _grokPattern = grokPattern;
+        _patterns = new Dictionary<string, string>();
+        _typeMaps = new Dictionary<string, string>();
+    }
 
-        public Grok(string grokPattern)
+    public GrokResult Parse(string text)
+    {
+        if (_compiledRegex == null)
```
Comment (Marusyk) [inline src/Grok.Net/Grok.cs:33]: You can use `is null` instead `== null`

File: src/Grok.Net/GrokItem.cs [hunk 3]
```diff
@@ -1,14 +1,13 @@
-﻿namespace GrokNet
+﻿namespace GrokNet;
+
+public sealed class GrokItem
```
Comment (Marusyk) [inline src/Grok.Net/GrokItem.cs:3]: Please convert it to a record

**Review by Marusyk — CHANGES_REQUESTED** [review]
LGTM but small fixes are required

File: src/Grok.Net/Grok.Net.csproj [hunk 4]
```diff
@@ -11,18 +10,22 @@
...
   </ItemGroup>
-
   <ItemGroup>
     <EmbeddedResource Include="grok-patterns" />
   </ItemGroup>
-
+  <ItemGroup>
+    <PackageReference Include="IsExternalInit" Version="1.0.1">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
```
[the remark this answers is not in the document]
Comment (vabka) [reply src/Grok.Net/Grok.Net.csproj:28]: .net 6 is now stable.
Maybe just drop .netstandard and leave .net6 only?

[the remark this answers is not in the document]
Comment (Marusyk) [reply src/Grok.Net/Grok.Net.csproj:28]: in that case, the package won't available for .net framework applications

**Comment by Marusyk** [comment]
I'm going to close this PR due to inactivity for a long time

**Pull request closed without merging**